platform/platform_freebsd.cpp

```diff
@@ -0,0 +1,142 @@
+#include "platform/platform.hpp"
+
+#include "base/logging.hpp"
+#include "coding/file_reader.hpp"
+
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/sysctl.h>
+
+
+/// @return directory where binary resides, including slash at the end
+static bool GetBinaryFolder(string & outPath)
```

Comment:
(replying to random comment)

This file is based on platform_linux.cpp, and it actually feels quite strange to be asked for style fixes in the code I haven't touched and to follow style upstream code doesn't. Will use common linux/freebsd code and fix bits that can be fixes though.